Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -374,6 +389,38 @@ def _validate_workspace_names(workspaces):
         return AnalysisDataService.Instance().retrieveWorkspaces(workspaces, unrollGroups=True)
 
 
+def _do_single_plot_mdhisto_workspace(ax, workspace, errors=False):
+    """Plot IMDHistoWorkspace
+    :param ax:
+    :param workspace:
+    :return:
+    """
+    from mantid.api import IMDHistoWorkspace
+
+    # Check it is a 1D IMDHistoWorkspace
+    if not isinstance(workspace, IMDHistoWorkspace) or workspace.getNumDims() > 1:
+        raise RuntimeError(f'Workspace {str(workspace)} is not an IMDHistoWorkspace or '
+                           f'its dimension is larger than 1')
+
+    # Get X and Y to plot
+    dim_x = workspace.getXDimension()
+    bins = np.arange(dim_x.getNBoundaries()) * dim_x.getBinWidth() + dim_x.getMinimum()
+    vec_x = 0.5 * (bins[1:] + bins[:-1])
+    vec_y = workspace.getSignalArray()
+
+    label_x = dim_x.getName()
+    label_y = 'Signal'
+
+    # Define plot function
+    plot_fn = ax.errorbar if errors else ax.plot
+
+    # Plot
+    ax.set_xlabel(label_x)
+    ax.set_ylabel(label_y)
+    plot_fn(vec_x, vec_y)
```

Comment:
There is no need to extract the data and the axes coordinates. Mantid can do that. Also, you can have the first non-integrated dimension to be something other than the XDimension